src/main/scala/model/persistence/Id.scala

```diff
@@ -69,19 +65,10 @@ object Id extends IdImplicitLike {
 }
 
 case class Id[T: IdType](value: T) extends HasValue[T] {
-  def isEmpty[A: TypeTag]: Boolean = value match {
-    case _ if typeOf[A] <:< typeOf[String]       => value == Id.empty[String]
-    case _ if typeOf[A] <:< typeOf[Long]         => value == Id.empty[Long]
-    case _ if typeOf[A] <:< typeOf[Option[Long]] => value == Id.empty[Option[Long]]
-    case _ if typeOf[A] <:< typeOf[UUID]         => value == Id.empty[UUID]
-  }
-
   override def toString: String = value.toString
 }
 
-trait HasId extends IdImplicitLike {
+abstract class HasId[A: IdType] extends IdImplicitLike {
```

Comment:
this requires a class not a trait now, because in order to provide the IdType, HasId needs to support argument lists, which traits do not. If you don't need ` IdType[A].emptyValue` in here, we can remove the context bound and make it a trait again.